Common/CUDA/tvdenoising.cu

```diff
@@ -628,3 +648,27 @@ void checkFreeMemory(const GpuIds& gpuids,size_t *mem_GPU_global){
         
         //*mem_GPU_global= insert your known number here, in bytes.
 }
+
+void cpy_from_host(float* device_array,float* host_array, 
+                   unsigned long long bytes_device,unsigned long long offset_device,unsigned long long offset_host, 
+                   unsigned long long pixels_per_slice, unsigned int buffer_length, 
+                   cudaStream_t stream, bool is_first_chunk, bool is_last_chunk,const long* image_size)
+{
+
+    // Initial and last cases are special. These define the boundary condition. In our case, we are using Neumann boundary condition
+    // so we need to copy the edge slice into the buffer
+    if(is_first_chunk){
+        for (unsigned int j=0;j<buffer_length;j++){
+            cudaMemcpyAsync(device_array+pixels_per_slice*j, host_array+pixels_per_slice*j, pixels_per_slice*sizeof(float), cudaMemcpyHostToDevice,stream);

```

Comment:
It would be batter to using the mirror slice as the buffer slice; here is the new code:
```
cudaMemcpyAsync(device_array+pixels_per_slice*j, host_array+pixels_per_slice*(buffer_length-j), pixels_per_slice*sizeof(float), cudaMemcpyHostToDevice,stream);
```